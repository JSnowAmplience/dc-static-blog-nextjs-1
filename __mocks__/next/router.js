module.exports = {
  useRouter: function() {
    return { asPath: 'mocked-path' };
  }
};
