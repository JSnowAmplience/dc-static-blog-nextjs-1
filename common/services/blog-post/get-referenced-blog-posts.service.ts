import { BlogPostReference } from '../../interfaces/blog-reference-list.interface';
import { getBlogPostByDeliveryKey } from '../blog-post.service';
import BlogPost from '../../interfaces/blog-post.interface';
// eslint-disable-next-line
const allSettled = require('promise.allsettled');

export default async function getReferencedBlogPosts(
  blogPosts: BlogPostReference[],
  stagingEnvironment?: string
): Promise<BlogPost[]> {
  const promises = blogPosts.map(
    (reference: BlogPostReference): Promise<BlogPost> =>
      getBlogPostByDeliveryKey(reference.deliveryKey, stagingEnvironment)
  );
  const promiseResults = await allSettled(promises);
  const rejectedPromises = promiseResults.filter(
    (promise: { status: string }): boolean => promise.status === 'rejected'
  );
  rejectedPromises.forEach((rejectedBlog: { reason: string }): void => console.warn(`Warn: ${rejectedBlog.reason}`));
  const hydratedBlogPosts = promiseResults
    .filter((promise: { status: string }): boolean => promise.status === 'fulfilled')
    .map((resolvedPromise: { value: BlogPost }): BlogPost => resolvedPromise.value);
  return hydratedBlogPosts;
}
