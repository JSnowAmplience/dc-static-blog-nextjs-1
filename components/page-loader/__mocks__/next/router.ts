export const mockRouterOn = jest.fn();
export const Router = { events: { on: mockRouterOn } };
export const useRouter = (): { asPath: string; query: {} } => ({ asPath: 'mocked-path', query: {} });
