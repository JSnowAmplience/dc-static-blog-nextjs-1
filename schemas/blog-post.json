{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "id": "https://raw.githubusercontent.com/amplience/dc-static-blog-nextjs/master/schemas/blog-post.json",
  "title": "Blog post",
  "description": "A bloblog post",
  "allOf": [
    {
      "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content"
    }
  ],
  "type": "object",
  "properties": {
    "title": {
      "title": "Title",
      "description": "Used for heading and SEO title tag",
      "type": "string",
      "minLength": 0,
      "maxLength": 150
    },
    "date": {
      "title": "Creation date",
      "description": "Creation date (YYYY-MM-DD)",
      "type": "string",
      "maxLength": 10
    },
    "description": {
      "title": "Description",
      "description": "Used for blog listing page and SEO description",
      "type": "string",
      "minLength": 0,
      "maxLength": 200
    },
    "image": {
      "title": "Image",
      "allOf": [
        {
          "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
        },
        {
          "properties": {
            "contentType": {
              "enum": [
                "https://raw.githubusercontent.com/amplience/dc-static-blog-nextjs/master/schemas/image.json"
              ]
            }
          }
        }
      ]
    },
    "urlSlug": {
      "title": "Url slug",
      "description": "Url friendly slug",
      "type": "string",
      "minLength": 0,
      "maxLength": 100,
      "pattern": "^[a-z0-9-]+$"
    },
    "tags": {
      "title": "Tags",
      "description": "Blog tags",
      "type": "array",
      "minItems": 0,
      "maxItems": 10,
      "items": {
        "type": "string",
        "minLength": 0,
        "maxLength": 100,
        "title": "Tag",
        "description": ""
      }
    },
    "readTime": {
      "title": "Read time",
      "description": "The time it takes to read the blog",
      "type": "integer",
    },
    "authors": {
      "title": "Blog author",
      "description": "Article author(s) - max 3",
      "type": "array",
      "min": 1,
      "max": 3,
      "items": {
        "allOf": [
          {
            "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
          },
          {
            "properties": {
              "contentType": {
                "enum": [
                  "https://raw.githubusercontent.com/amplience/dc-static-blog-nextjs/master/schemas/author.json"
                ]
              }
            }
          }
        ]
      }
    },
    "content": {
      "title": "Content",
      "description": "",
      "type": "array",
      "minItems": 0,
      "maxItems": 20,
      "items": {
        "allOf": [
          {
            "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content-link"
          },
          {
            "properties": {
              "contentType": {
                "title": "Content",
                "enum": [
                  "https://raw.githubusercontent.com/amplience/dc-static-blog-nextjs/master/schemas/image.json",
                  "https://raw.githubusercontent.com/amplience/dc-static-blog-nextjs/master/schemas/video.json",
                  "https://raw.githubusercontent.com/amplience/dc-static-blog-nextjs/master/schemas/text.json"
                ]
              }
            }
          }
        ]
      }
    }
  },
  "propertyOrder": [
    "title",
    "authors",
    "date",
    "description",
    "image",
    "urlSlug",
    "tags",
    "readTime",
    "content"
  ]
}
