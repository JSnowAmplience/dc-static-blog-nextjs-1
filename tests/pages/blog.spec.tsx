/* eslint-env jest */
import renderer from 'react-test-renderer';
import BlogPostPage from '../../pages/blog';
import blogPostFixture from '../fixtures/single-blog-post-data-object.json';
import { getBlogPostByDeliveryKey } from '../../common/services/blog-post.service';
import { NextPageContext } from 'next';

jest.mock('../../common/services/blog-post.service', () => {
  return {
    __esModule: true,
    getBlogPostByDeliveryKey: jest.fn()
  };
});

const mockGetBlogPostByDeliveryKey: jest.MockedFunction<typeof getBlogPostByDeliveryKey> = getBlogPostByDeliveryKey as jest.MockedFunction<
  typeof getBlogPostByDeliveryKey
>;

describe('Blog page', () => {
  beforeEach(() => {
    process.env.DYNAMIC_CONTENT_REFERENCE_ID = 'reference-id';
    process.env.GA_TRACKING_ID = 'ga-tracking-id';
    jest.clearAllMocks();
  });

  test('renders Blog page with content', () => {
    const component = renderer.create(<BlogPostPage blogPost={blogPostFixture} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('getInitialProps returns blog post content via api', async () => {
    mockGetBlogPostByDeliveryKey.mockResolvedValue(blogPostFixture);
    const query = { deliveryKey: 'blog-id' };
    const result = await BlogPostPage.getInitialProps({ query, pathname: '/blog' } as unknown as NextPageContext);
    expect(result).toEqual({ blogPost: blogPostFixture });
  });

  test('getInitialProps should call getHydratedBlogList with staging environment', async () => {
    mockGetBlogPostByDeliveryKey.mockResolvedValue(blogPostFixture);
    const query = { deliveryKey: 'blog-id', vse: 'vse-base-url' };
    await BlogPostPage.getInitialProps({ query, pathname: '/blog' } as unknown as NextPageContext);
    expect(mockGetBlogPostByDeliveryKey).toHaveBeenCalledWith(query.deliveryKey, `//${query.vse}`);
  });

  test('getInitialProps should call getHydratedBlogList without staging environment', async () => {
    mockGetBlogPostByDeliveryKey.mockResolvedValue(blogPostFixture);
    const query = { deliveryKey: 'blog-id' };
    await BlogPostPage.getInitialProps({ query, pathname: '/blog' } as unknown as NextPageContext);
    expect(mockGetBlogPostByDeliveryKey).toHaveBeenCalledWith(query.deliveryKey, undefined);
  });

  test('getInitialProps throws error when deliveryKey is undefined', async () => {
    mockGetBlogPostByDeliveryKey.mockRejectedValue(new Error());
    const query = {};
    await expect(BlogPostPage.getInitialProps({ query, pathname: '/blog' } as unknown as NextPageContext)).rejects.toThrowError(Error);
  });
});
