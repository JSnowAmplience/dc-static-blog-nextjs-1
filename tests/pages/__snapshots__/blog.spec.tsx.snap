// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog page renders Blog page with content 1`] = `
<div
  className="jsx-873644259 site-container"
>
  <section
    className="jsx-1255100151"
  >
    <header
      className="jsx-1255100151"
    >
      <a
        aria-label=""
        className="jsx-3354872520"
        href="/"
      >
        <img
          alt="Amplience logo"
          className="jsx-1701653183 logo"
          src="/static/images/logo-white-340w.png"
          srcSet="/static/images/logo-white-160w.png 1x, /static/images/logo-white-340w.png 2x"
        />
      </a>
      <h1
        className="jsx-1255100151"
      >
        PRODUCT BLOG
      </h1>
    </header>
  </section>
  <div
    className="jsx-873644259 site-content"
  >
    <main
      className="jsx-873644259"
    >
      <div
        className="jsx-2814946630 content-header"
      >
        <section
          className="jsx-3378232041"
        >
          <div
            className="jsx-3378232041 avatars"
          >
            <div
              className="jsx-2854206183"
            >
              <div
                className="jsx-2854206183 avatar"
              >
                <span
                  className="lazy-img"
                  style={
                    Object {
                      "display": "inline-block",
                      "height": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="jsx-3378232041 publish-data"
          >
            <div
              className="jsx-3378232041 authors"
            >
              <div
                className="jsx-3680733511 name"
              >
                Sir Blog
              </div>
            </div>
            <div
              className="jsx-3378232041 publish-metadata"
            >
              <div
                className="jsx-3378232041 date"
              >
                August 07, 2019
              </div>
              <div
                className="jsx-3378232041 readTime"
              >
                12
                 mins read
              </div>
            </div>
          </div>
        </section>
        <section
          className="jsx-3528769338"
        >
          <h1
            className="jsx-2799736912"
          >
            My First Blog Post
          </h1>
          <h2
            className="jsx-1054465095"
          >
            Some text for the description that explains this blog post
          </h2>
        </section>
      </div>
      <div
        className="jsx-2814946630 blog-image"
      >
        <span
          className="lazy-img"
          style={
            Object {
              "display": "inline-block",
              "height": 0,
              "width": 0,
            }
          }
        />
      </div>
      <div
        className="jsx-2814946630 content-wrapper"
      >
        <section
          className="jsx-2988077148"
        >
          <div
            className="jsx-2988077148"
          >
            <span
              className="lazy-img"
              style={
                Object {
                  "display": "inline-block",
                  "height": 0,
                  "width": 0,
                }
              }
            />
          </div>
          <div
            className="jsx-2988077148"
          >
            <h1
              className="jsx-2595089998"
            >
              This is my first Blog
            </h1>
            <h2
              className="jsx-2595089998"
            >
              Look-a-header 2
            </h2>
            <h3
              className="jsx-2595089998"
            >
              Look-a-header 3
            </h3>
            <h4
              className="jsx-2595089998"
            >
              Look-a-header 4
            </h4>
            <h5
              className="jsx-2595089998"
            >
              Look-a-header 5
            </h5>
            <h6
              className="jsx-2595089998"
            >
              Look-a-header 6
            </h6>
            <p
              className="jsx-430790654"
            >
              <em>
                <strong>
                  This
                </strong>
              </em>
               
              <em>
                is
              </em>
               
              <em>
                some
              </em>
               
              <del>
                sample
              </del>
               
              <strong>
                text
              </strong>
              .
            </p>
            <p
              className="jsx-430790654"
            >
              <img
                alt="alt text"
                src="https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png"
                title="Logo Title Text 1"
              />
            </p>
            <p
              className="jsx-430790654"
            >
              This is some more sample text with a 
              <span
                className="jsx-3731109703"
              >
                <a
                  className="jsx-3655793200"
                  href="http://bbc.co.uk"
                  rel="noreferrer"
                  target="_new"
                >
                  LINK
                </a>
              </span>
            </p>
            <blockquote
              className="jsx-2064009043"
            >
              <p
                className="jsx-430790654"
              >
                Comments are cool!
              </p>
              <p
                className="jsx-430790654"
              >
                See what I mean
              </p>
            </blockquote>
            <p
              className="jsx-430790654"
            >
              Look some code:
            </p>
            <pre
              style={
                Object {
                  "MozHyphens": "none",
                  "MozTabSize": "4",
                  "OTabSize": "4",
                  "WebkitHyphens": "none",
                  "background": "#1d1f21",
                  "borderRadius": "0.3em",
                  "color": "#c5c8c6",
                  "direction": "ltr",
                  "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
                  "hyphens": "none",
                  "lineHeight": "1.5",
                  "margin": ".5em 0",
                  "msHyphens": "none",
                  "overflow": "auto",
                  "padding": "1em",
                  "tabSize": "4",
                  "textAlign": "left",
                  "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                  "whiteSpace": "pre",
                  "wordBreak": "normal",
                  "wordSpacing": "normal",
                }
              }
            >
              <code
                style={
                  Object {
                    "MozHyphens": "none",
                    "MozTabSize": "4",
                    "OTabSize": "4",
                    "WebkitHyphens": "none",
                    "color": "#c5c8c6",
                    "direction": "ltr",
                    "float": "left",
                    "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
                    "hyphens": "none",
                    "lineHeight": "1.5",
                    "msHyphens": "none",
                    "paddingRight": "10px",
                    "tabSize": "4",
                    "textAlign": "left",
                    "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                    "whiteSpace": "pre",
                    "wordBreak": "normal",
                    "wordSpacing": "normal",
                  }
                }
              >
                <span
                  className="react-syntax-highlighter-line-number"
                  style={Object {}}
                >
                  1

                </span>
                <span
                  className="react-syntax-highlighter-line-number"
                  style={Object {}}
                >
                  2

                </span>
              </code>
              <code
                style={
                  Object {
                    "MozHyphens": "none",
                    "MozTabSize": "4",
                    "OTabSize": "4",
                    "WebkitHyphens": "none",
                    "color": "#c5c8c6",
                    "direction": "ltr",
                    "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
                    "hyphens": "none",
                    "lineHeight": "1.5",
                    "msHyphens": "none",
                    "tabSize": "4",
                    "textAlign": "left",
                    "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                    "whiteSpace": "pre",
                    "wordBreak": "normal",
                    "wordSpacing": "normal",
                  }
                }
              >
                <span
                  style={Object {}}
                >
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#96CBFE",
                      }
                    }
                  >
                    const
                  </span>
                  <span
                    style={Object {}}
                  >
                     something 
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#EDEDED",
                      }
                    }
                  >
                    =
                  </span>
                  <span
                    style={Object {}}
                  >
                     
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#A8FF60",
                      }
                    }
                  >
                    'anything'
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    ;
                  </span>
                  <span
                    style={Object {}}
                  >
                    

                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    style={Object {}}
                  >
                    
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#DAD085",
                      }
                    }
                  >
                    calls
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    (
                  </span>
                  <span
                    style={Object {}}
                  >
                    something
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    )
                  </span>
                  <span
                    className="token"
                    style={
                      Object {
                        "color": "#c5c8c6",
                      }
                    }
                  >
                    ;
                  </span>
                </span>
              </code>
            </pre>
            <p
              className="jsx-430790654"
            >
              inline 
              <code
                className="jsx-3800523359"
              >
                code
              </code>
               is awesome underlined
            </p>
            <hr />
            &lt;h4&gt;A title&lt;/h4&gt;
            <p
              className="jsx-430790654"
            >
              A table too:
            </p>
            <table>
              <thead>
                <tr>
                  <th>
                    Hello
                  </th>
                  <th>
                    world
                  </th>
                  <th>
                    table
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
                <tr>
                  <td>
                    some
                  </td>
                  <td>
                    values
                  </td>
                  <td>
                    here
                  </td>
                </tr>
              </tbody>
            </table>
            <p
              className="jsx-430790654"
            >
              Get listed
            </p>
            <ul>
              <li>
                Item One
              </li>
              <li>
                Item Two
              </li>
              <li>
                Item Three
              </li>
            </ul>
          </div>
          <div
            className="jsx-2988077148 blog-post-video"
          >
            
          </div>
        </section>
      </div>
      <div
        className="jsx-3936729912 content-footer"
      >
        <section
          className="jsx-2442235184"
        >
          <h3
            className="jsx-2442235184"
          >
            Share the post
          </h3>
          <div
            className="jsx-2442235184"
          >
            <div
              className="jsx-2442235184"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<script async src=\\"https://platform.linkedin.com/in.js\\" type=\\"text/javascript\\">lang: en_US</script><script type=\\"IN/Share\\" data-url=mocked-path></script>",
                }
              }
              suppressHydrationWarning={true}
            />
            <div
              className="jsx-2442235184"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<a href=\\"https://twitter.com/share?ref_src=twsrc%5Etfw\\" class=\\"twitter-share-button\\" data-text=My First Blog Post data-show-count=\\"false\\">Tweet</a><script async src=\\"https://platform.twitter.com/widgets.js\\" charSet=\\"utf-8\\"></script>",
                }
              }
              suppressHydrationWarning={true}
            />
          </div>
        </section>
      </div>
    </main>
    <footer
      className="jsx-27833366"
    >
      <section
        className="jsx-27833366"
      >
        <div
          className="jsx-27833366 footer-info"
        >
          <div
            className="jsx-27833366 footer-info-top"
          >
            <img
              alt="Amplience logo"
              className="jsx-1701653183 logo"
              src="/static/images/logo-340w.png"
              srcSet="/static/images/logo-160w.png 1x, /static/images/logo-340w.png 2x"
            />
            <div
              className="jsx-27833366 social-links"
            >
              <a
                className="jsx-3655793200"
                href="//twitter.com/amplience"
                rel="noreferrer"
                target="_new"
              >
                <img
                  alt="Twitter logo"
                  className="jsx-27833366"
                  src="/static/images/social/ic_twitter.svg"
                />
              </a>
              <a
                className="jsx-3655793200"
                href="//www.linkedin.com/company/amplience/"
                rel="noreferrer"
                target="_new"
              >
                <img
                  alt="Linkedin logo"
                  className="jsx-27833366"
                  src="/static/images/social/ic_linkedin.svg"
                />
              </a>
            </div>
          </div>
          <div
            className="jsx-27833366 copyright"
          >
            © Copyright Amplience 
            2019
          </div>
        </div>
        <div
          className="jsx-27833366 footer-links"
        >
          <section
            className="jsx-287747732 footer-links-section"
          >
            <h4
              className="jsx-287747732"
            >
              For Developers
            </h4>
            <ul
              className="jsx-287747732"
            >
              <li>
                <a
                  className="jsx-3655793200"
                  href="//github.com/amplience/"
                  rel="noreferrer"
                  target="_new"
                >
                  GitHub
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//stackoverflow.com/questions/tagged/amplience-dynamic-content"
                  rel="noreferrer"
                  target="_new"
                >
                  Stack Overflow
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/integration/developergettingstarted.html"
                  rel="noreferrer"
                  target="_new"
                >
                  Dynamic Content docs
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/contenthub.html"
                  rel="noreferrer"
                  target="_new"
                >
                  Content Hub docs
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/integration/reference.html"
                  rel="noreferrer"
                  target="_new"
                >
                  API reference
                </a>
              </li>
            </ul>
          </section>
          <section
            className="jsx-287747732 footer-links-section"
          >
            <h4
              className="jsx-287747732"
            >
              Company
            </h4>
            <ul
              className="jsx-287747732"
            >
              <li>
                <a
                  className="jsx-3655793200"
                  href="//amplience.com"
                  rel="noreferrer"
                  target="_new"
                >
                  Website
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//amplience.com/careers"
                  rel="noreferrer"
                  target="_new"
                >
                  Careers
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//amplience.com/privacy-cookie-policy"
                  rel="noreferrer"
                  target="_new"
                >
                  Privacy policy
                </a>
              </li>
              <li>
                <a
                  className="jsx-3655793200"
                  href="//docs.amplience.net/releasenotes/readme.html"
                  rel="noreferrer"
                  target="_new"
                >
                  What\`s new
                </a>
              </li>
            </ul>
          </section>
        </div>
      </section>
    </footer>
    <div
      className="cookieConsent "
      style={
        Object {
          "alignItems": "center",
          "background": "rgba(255, 255, 255, 0.9)",
          "bottom": "0",
          "boxShadow": "20px 20px 20px 10px #000",
          "color": "#333",
          "flexDirection": "column",
          "position": "fixed",
          "textAlign": "center",
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        className="cookie-banner-content"
        style={
          Object {
            "padding": "20px 0px 10px",
          }
        }
      >
        We use cookies to give you the best possible experience on our website. By continuing to browse this site, you give consent for cookies to be used. For more details please read our
         
        <a
          href="https://amplience.com/privacy-cookie-policy/"
          rel="noreferrer"
          target="_blank"
        >
          Cookie Policy
        </a>
        .
      </div>
      <button
        className="cookie-banner-btn"
        id=""
        onClick={[Function]}
        style={
          Object {
            "background": "#fff",
            "border": "3px solid #CCC",
            "borderRadius": "5px",
            "color": "#333",
            "marginBottom": "10px",
          }
        }
      >
        Accept Cookies
      </button>
    </div>
    <script
      async={true}
      src="https://www.google-analytics.com/analytics.js"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window.ga=window.ga||function(){(ga.q = ga.q || []).push(arguments)};ga.l=+new Date; ga('create', 'ga-tracking-id', 'auto'); ga('send', 'pageview');",
        }
      }
    />
  </div>
</div>
`;
